pkg/iac/scanners/kubernetes/parser/parser.go

```diff
@@ -5,69 +5,14 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"io/fs"
-	"path/filepath"
 	"regexp"
 	"strings"
 
 	"gopkg.in/yaml.v3"
 	kyaml "sigs.k8s.io/yaml"
-
-	"github.com/aquasecurity/trivy/pkg/log"
 )
 
-type Parser struct {
-	logger *log.Logger
-}
-
-// New creates a new K8s parser
-func New() *Parser {
-	return &Parser{
-		logger: log.WithPrefix("k8s parser"),
-	}
-}
-
-func (p *Parser) ParseFS(ctx context.Context, target fs.FS, path string) (map[string][]any, error) {
-	files := make(map[string][]any)
-	if err := fs.WalkDir(target, filepath.ToSlash(path), func(path string, entry fs.DirEntry, err error) error {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
-		if err != nil {
-			return err
-		}
-		if entry.IsDir() {
-			return nil
-		}
-
-		parsed, err := p.ParseFile(ctx, target, path)
-		if err != nil {
-			p.logger.Error("Parse error", log.FilePath(path), log.Err(err))
-			return nil
-		}
-
-		files[path] = parsed
-		return nil
-	}); err != nil {
-		return nil, err
-	}
-	return files, nil
-}
-
-// ParseFile parses Kubernetes manifest from the provided filesystem path.
-func (p *Parser) ParseFile(_ context.Context, fsys fs.FS, path string) ([]any, error) {
-	f, err := fsys.Open(filepath.ToSlash(path))
-	if err != nil {
-		return nil, err
-	}
-	defer func() { _ = f.Close() }()
-	return p.Parse(f, path)
-}
```

Comment:
Were they not used? I'm a little skeptical because they are exported funcs which may have been used elsewhere (outside of the project).

Reply:
This parser is specific to Trivy, I don't think it can be used anywhere else.

Reply:
OK - I guess only one way to find out 😺